renderer/HardwareBaseGPU.cpp

```diff
@@ -528,6 +476,26 @@ void rend_DrawPolygon2D(int handle, g3Point **p, int nv) {
   gpu_RenderPolygon(&vArray[0], nv);
 }
 
+color_array DeterminePointColor(g3Point const* pnt, bool disableGouraud, bool checkTextureQuality, bool flatColorForNoLight) {
+  auto alpha = gpu_Alpha_multiplier * gpu_Alpha_factor;
+  if (gpu_state.cur_alpha_type & ATF_VERTEX) {
+    alpha *= pnt->p3_a;
+  }
+
+  // If we have a lighting model, apply the correct lighting!
+  if ((gpu_state.cur_light_state == LS_FLAT_GOURAUD && !disableGouraud) ||
```

Comment:
Maybe we should create templates and specializations of this function to replace branching by static dispatch to increase performance

Reply:
ooh, great idea. if i'm not able to pretty shortly turn this into shader logic, i'll convert it to template logic.